src/truncate.jl

```diff
@@ -61,6 +60,29 @@ logccdf(d::Truncated, x::Float64) = x <= d.lower ? 0.0 :
 quantile(d::Truncated, p::Float64) = quantile(d.untruncated, d.lcdf + p * d.tp)
 
 
+pdf(d::Truncated, x::Int64) = d.lower <= x <= d.upper ? pdf(d.untruncated, x) / d.tp : 0.0
+
+logpdf(d::Truncated, x::Int64) = d.lower <= x <= d.upper ? logpdf(d.untruncated, x) - d.logtp : -Inf
+
+cdf(d::Truncated, x::Int64) = x <= d.lower ? 0.0 : 
+                             x >= d.upper ? 1.0 :
+                             (cdf(d.untruncated, x) - d.lcdf) / d.tp
+
+logcdf(d::Truncated, x::Int64) = x <= d.lower ? -Inf :
+                                x >= d.upper ? 0.0 :
+                                log(cdf(d.untruncated, x) - d.lcdf) - d.logtp
+
+ccdf(d::Truncated, x::Int64) = x <= d.lower ? 1.0 : 
+                              x >= d.upper ? 0.0 :
+                              (d.ucdf - cdf(d.untruncated, x)) / d.tp
+
+logccdf(d::Truncated, x::Int64) = x <= d.lower ? 0.0 :
+                                 x >= d.upper ? -Inf :
+                                 log(d.ucdf - cdf(d.untruncated, x)) - d.logtp 
+
+
+quantile(d::Truncated, p::Int64) = quantile(d.untruncated, d.lcdf + p * d.tp)
+
```

Comment:
Ah, I see. What about if you made them a union type instead? If that still causes problems, you can leave them as they are.

Also, you would probably want `Int`, instead of `Int64` (that way it will work on 32bit machines). And you don't need the quantile methods (since integer arguments don't really make sense, the usual promotion machinery for them is fine)